app/code/Magento/SalesRule/Controller/Adminhtml/Promo/Quote/Save.php

```diff
@@ -65,11 +65,11 @@ public function execute()
             try {
                 /** @var $model \Magento\SalesRule\Model\Rule */
                 $model = $this->_objectManager->create(\Magento\SalesRule\Model\Rule::class);
+                $data = $this->getRequest()->getPostValue();
```

Comment:
I guess it would be even better to move the `$data` declaration out of the `try-catch` block. :)
e.g.
```
public function execute()
{
    $data = $this->getRequest()->getPostValue();
    if ($data) {
            try {
                /** @var $model \Magento\SalesRule\Model\Rule */
                $model = $this->_objectManager->create(\Magento\SalesRule\Model\Rule::class);
                $this->_eventManager->dispatch(
                    'adminhtml_controller_salesrule_prepare_save',
                    ['request' => $this->getRequest()]
                );
   ...
```

Reply:
Hi @dmytro-ch 
I have updated the code as per your suggestion.